modules/api/pkg/ee/clusterbackup/storage-location/handler.go

```diff
@@ -194,20 +194,35 @@ func ListCSBLBucketObjects(ctx context.Context, request interface{}, userInfoGet
 	return provider.ListBucketObjects(ctx, user, req.ClusterBackupStorageLocationName, labelSet)
 }
 
-func CreateCBSL(ctx context.Context, request interface{}, userInfoGetter provider.UserInfoGetter, provider provider.BackupStorageProvider, projectProvider provider.ProjectProvider) (*apiv2.ClusterBackupStorageLocation, error) {
+func CreateCBSL(
+	ctx context.Context,
+	request interface{},
+	userInfoGetter provider.UserInfoGetter,
+	provider provider.BackupStorageProvider,
+	projectProvider provider.ProjectProvider,
+	settingsProvider provider.SettingsProvider,
+) (*apiv2.ClusterBackupStorageLocation, error) {
 	req, ok := request.(createCbslReq)
 	if !ok {
 		return nil, utilerrors.NewBadRequest("invalid request")
 	}
 
 	user, err := userInfoGetter(ctx, req.ProjectID)
+	if err != nil {
+		return nil, err
+	}
 
+	settings, err := settingsProvider.GetGlobalSettings(ctx)
 	if err != nil {
 		return nil, err
 	}
 
+	if settings == nil {
```

Comment:
I think this check is not necessary but if you want to keep it then error message should be updated because these are global settings are not project settings.

Reply:
Thanks, updated in 6dc87863da980e4b47a6226f0d1c509d593feee6